File: frontend/models/Timetable.ts
export interface TimetableService {
    id: number;
    line_name: string;
    inbound: boolean;
}

export interface TimetableStop {
    id: string;
    name: string;
    timing_status: string;
}

export interface TimetableJourney {
    id: number;
    start_time: string;
    times: (string | null)[];
}

export interface TimetableResponse {
    service: TimetableService;
    stops: TimetableStop[];
    journeys: TimetableJourney[];
}

export interface DBTimetable {
    id: number;
    service_code: string;
    line_id: string;
    line_name: string;
    line_brand: string | null;
    service_id: number;
    bt_service_id: number | null;
    revision_number: number;
    description: string;
    outbound_description: string;
    inbound_description: string;
    origin: string;
    destination: string;
    vias: string | null;
    public_use: boolean;
    data_source_id: number;
    created_at: string;
    modified_at: string;
    start_date: string;
    end_date: string;
    journey_count?: number;
}

File: backend/schemas/timetable.py
from datetime import date, datetime
from pydantic import BaseModel, ConfigDict
from typing import Optional


class TimetableService(BaseModel):
    id: int
    line_name: str
    inbound: bool


class TimetableStop(BaseModel):
    id: str
    name: str
    timing_status: str


class TimetableJourney(BaseModel):
    id: int
    start_time: str
    times: list[Optional[str]]


class TimetableResponse(BaseModel):
    service: TimetableService
    stops: list[TimetableStop]
    journeys: list[TimetableJourney]


class DBTimetable(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    service_code: str
    line_id: str
    line_name: str
    line_brand: Optional[str] = None
    service_id: int
    bt_service_id: Optional[int] = None
    revision_number: int
    description: Optional[str]
    outbound_description: str
    inbound_description: str
    origin: str
    destination: str
    vias: Optional[str] = None
    public_use: bool
    data_source_id: int
    created_at: datetime
    modified_at: datetime
    start_date: date
    end_date: Optional[date] = None
    journey_count: Optional[int] = None
